ClientTests/Home/HomeHistoryHighlightsViewModelTests.swift

```diff
@@ -0,0 +1,91 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+
+@testable import Client
+
+class HomeHistoryHighlightsViewModelTests: XCTestCase {
+
+    private var profile: MockProfile!
+    private var tabManager: TabManager!
+    private var entryProvider: HistoryHighlightsTestEntryProvider!
+
+    override func setUp() {
+        super.setUp()
+
+        profile = MockProfile(databasePrefix: "historyHighlightViewModel_tests")
+        profile._reopen()
+        tabManager = TabManager(profile: profile, imageStore: nil)
+        entryProvider = HistoryHighlightsTestEntryProvider(with: profile, and: tabManager)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+
+        profile._shutdown()
+        profile = nil
+        tabManager = nil
+        entryProvider = nil
+    }
+
+    func testViewModelCreation_WithNoEntries() {
+        entryProvider.emptyDB()
+        let viewModel = FxHomeHistoryHightlightsViewModel(with: profile,
+                                                          isPrivate: false,
+                                                          tabManager: tabManager)
+        delayTest {
```

Comment:
to have more reliable test I think is best to add the completion to FxHomeHistoryHightlightsViewModel init